mainwindow.cpp

```diff
@@ -799,6 +804,11 @@ void MainWindow::showJohnFinished(int exitCode, QProcess::ExitStatus exitStatus)
     m_ui->actionStart_Attack->setEnabled(true);
     m_ui->actionOpen_Password->setEnabled(true);
     verifySessionState();
+
+    if (exitStatus == QProcess::CrashExit) {
```

Comment:
When you handled exitcode stuff in the PR todo, I think that a bug was introduced by the return; because when john crash, the button stay disabled forever and the user can't retry to start/resume an attack or open another file. In fact, he is blocked and can't do anything useful so I changed that.